WordPress/gutenberg
# Improves the UX of menu management in the navigation block

## What?
Closes #42257, #42602,  #38169

## Why?
Simplifying the UX of the navigation block. The crowding of the block toolbar with all kinds of disconnected actions does not appear as a good path. 

## How?
A few tweaks:

- [x] move the items in the "select menu" block control to the block's inspector as follows:
  - [x] the list of menus as a dropdown in an inspector panel
  - [X] the manage menus action as a link button below the selector

**Optional:**
- [ ] add a list view representing the tree of InnerBlocks of the navigation block to the block's inspector
  - [ ] allow adding new menu items from the list view representation
  - [ ] disable inner block selection from the list view
  - [ ] cull some of the useless list view item options (e.g. make template part)


**To do**

- [x] There has to be a chevron link in the dropdown toggle
- [x] The label of the currently selected menu does not update on change
- [x] The label of the currently selected menu does not update if menu is renamed from advanced
- [x] Use a visually hidden label to explain this is a menu switcher
- [x] Default to a button if there are no block menus and no classic menus
- [x] Rename "Loading options ..." to "Loading ..."
- [x] Rename "Select another menu" to "Select menu"
- [x] Removed [disabled](https://github.com/WordPress/gutenberg/pull/42987#discussion_r962116873) and implement `isBusy`
- [ ] Update the navigation tests according to the new UI of the block

## Testing Instructions
1. Check out this PR
2. Add a navigation block
3. Behold the amazing UX
4. Test that:

- the menu can be switched to another menu
- importing classic menus work
- new menus can be created
- renaming menus in advanced is immediately reflected in the control
- keyboard navigation works as expected
- screen readers announce this control properly

## Screenshots or screencast 

https://user-images.githubusercontent.com/107534/187418826-c999c81d-64f2-42d5-97c1-063b149ce367.mp4
